features/machine/machine_healthcheck.feature

```diff
@@ -260,19 +260,44 @@ Feature: MachineHealthCheck Test Scenarios
     Then the output should contain:
       | mhc-<%= machine_set.name %>: total targets: 1,  maxUnhealthy: 1%, unhealthy: 1. Short-circuiting remediation |
     """
-
+    
   # @author miyadav@redhat.com
   # @case_id OCP-33714
   @admin
-  Scenario: Leverage OpenAPI validation within MHC
+  Scenario Outline: Leverage OpenAPI validation within MHC
     Given I have an IPI deployment
     And I switch to cluster admin pseudo user
     Then I use the "openshift-machine-api" project
 
-    # Create MHC with malformed unhealthy nodes value and empty selectors
-    Given I obtain test data file "cloud/mhc/mhc_malformed.yaml"
-    When I run the :create client command with:
-      | f | mhc_malformed.yaml | 
+    Given I obtain test data file "cloud/mhc/mhc_validations.yaml"
+    When I run oc create over "mhc_validations.yaml" replacing paths:
+      | ["spec"]["maxUnhealthy"] | <maxUnhealthy> |
+    Then the output should contain:
+      | maxUnhealthy: Invalid value: "": spec.maxUnhealthy in body should match '^((100|[0-9]{1,2})%|[0-9]+)$ |
+
+   Examples:
+      | maxUnhealthy |
+      |  -2a         |
+      | "10t%"       |
+      | "-2%"        |
+      |  -2%         |
+
+  # @author miyadav@redhat.com
+  # @case_id OCP-34095
+  @admin
+  Scenario Outline: [mhc] timeout field without units(h,m,s) shoud not be allowed to be stored
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+    Then I use the "openshift-machine-api" project
+
+    Given I obtain test data file "cloud/mhc/mhc_validations.yaml"
+    When I run oc create over "mhc_validations.yaml" replacing paths:
+      | ["spec"]["unhealthyConditions"][0]["timeout"] | <timeout> |
     Then the output should contain:
-      | The MachineHealthCheck "mhc-malformed" is invalid: |
+      | timeout: Invalid value: "": spec.unhealthyConditions.timeout in body should match '^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$'|
```

Comment:
@miyadav do you think the output message match is too verbose?

Reply:
@pruan-rht , yes , but the way it is developed , it will be like that and we recently found few bugs around that message , hence used it . Do you feel we should keep only "timeout: Invalid value: "": spec.unhealthyConditions.timeout" ?

Reply:
made the change